public/js/app.js

```diff
@@ -0,0 +1,149 @@
+'use strict';
+
+let lastSubreddit = '';
+
+function getData(subreddit) {
+  lastSubreddit = subreddit;
+  let posts = '';
+
+  function subRedditErrorHandler() {
+    let updateDefaultError = document.querySelector('.errorNotice');
+    updateDefaultError.style.display = 'block';
+  }
+
+  function subRedditReqListener() {
+    content.innerHTML = '';
+
+    posts = JSON.parse(this.responseText).data.children;
```

Comment:
For this line, try:

```
const data = JSON.parse(this.responseText).data;
```

then your if can be:

```
if (!data) { 
  ...what you already have in your if body
} else {
  posts = data.children;
}
```

Also, make sure you add a return at the end of your if body so that none of the code below the if gets executed.